controller/k8s/api_test.go

```diff
@@ -879,8 +879,8 @@ status:
 				t.Fatalf("api.GetPodsFor() unexpected error, expected [%s] got: [%s]", exp.err, err)
 			}
 
-			sort.Sort(byPod(pods))
-			sort.Sort(byPod(k8sResultPods))
+			sort.Sort(podByUID(pods))
+			sort.Sort(podByUID(k8sResultPods))
```

Comment:
i think this will only sort the new slice created by `podByUID`, and leave the original ones, that get passed into `DeepEqual`, unchanged.

Reply:
you're right, since it returns a new array, the `sort.Sort` function just sort it and let the original one as it's.

Reply:
I could see test running for multiple times without flakiness with these changes: https://gist.github.com/jonathanbeber/33adb4bb051142eb2a30e24e938933ba